orderBook.js

```diff
@@ -1,5 +1,34 @@
 function reconcileOrder(existingBook, incomingOrder) {
-  throw new Error('Not Implemented Yet')
+  // If list is empty just add the order and return
+
+  let updatedBook = []
+
+  if (existingBook.length === 0) {
+    existingBook.push(incomingOrder)
+    return existingBook
+  }
+
+  if ((existingBook[0].type === 'buy' && incomingOrder.type === 'sell') && (existingBook[0].quantity === incomingOrder.quantity) && (existingBook[0].price === incomingOrder.price)) {
+    return updatedBook
+  }
+
+  else if ((existingBook[0].type === 'buy' && incomingOrder.type === 'sell') && (existingBook[0].quantity > incomingOrder.quantity) && (existingBook[0].price === incomingOrder.price)) {
```

Comment:
Given that each of these ifs has a return I wouldn't even bother with the elses.